.github/workflows/main.yaml

```diff
@@ -30,13 +30,32 @@ jobs:
     with:
       repository: ${{ needs.vars.outputs.repository }}
       tag: ${{ needs.vars.outputs.tag }}
+      dockerfile: docker/Dockerfile.test
```

Comment:
Changing the dockerfile here will modify the build image that is deployed to dev, stage and prod. 

Tests in the run_unit_tests job are not really using this dockerfile. They run in the `ubuntu-latest` runner and install the python distribution and dependencies. 
I suggest deleting Dockerfile.test and continue using the normal dockerfile for the build step. Or use the image built in the step https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container

Reply:
I agree that whatever we do here for unit tests, shouldn't affect the images we build for dev/stage/prod. But the reason we added `Dockerfile.test` is that we need to build a different image to run unit tests, including extra steps to generate requirements.txt and then install dependencies, which are needed for running tests only. Maybe we need to define the workflow in a different way for running unit tests. Using docker multi-stage builds may help to re-use common layers here too.

Reply:
The funny thing is the tests are not using the docker image at all becausethe run_unit_tests job is not referencing it and it's installing python and the dependencies manually

Reply:
i can set a different step to build the test image (with tests dependencies), and use the real build for deployments. Do you think that is a good alternative?